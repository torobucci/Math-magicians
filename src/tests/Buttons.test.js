import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Buttons from '../components/Buttons';

describe('Buttons are rendered correctly', () => {
  test('Buttons snapshots match', () => {
    const tree = renderer.create(<Buttons />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('ensure all buttons are rendered', () => {
    render(<Buttons />);
    const buttons = screen.getAllByRole('button');
    expect(buttons.length).toBe(19);
  });
});
