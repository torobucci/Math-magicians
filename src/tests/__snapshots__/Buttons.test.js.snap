// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons are rendered correctly Buttons snapshots match 1`] = `
Array [
  <button
    className="button"
    type="button"
  >
    A/C
  </button>,
  <button
    className="button"
    type="button"
  >
    +/-
  </button>,
  <button
    className="button"
    type="button"
  >
    %
  </button>,
  <button
    className="button operator"
    type="button"
  >
    รท
  </button>,
  <button
    className="button"
    type="button"
  >
    7
  </button>,
  <button
    className="button"
    type="button"
  >
    8
  </button>,
  <button
    className="button"
    type="button"
  >
    9
  </button>,
  <button
    className="button operator"
    type="button"
  >
    x
  </button>,
  <button
    className="button"
    type="button"
  >
    4
  </button>,
  <button
    className="button"
    type="button"
  >
    5
  </button>,
  <button
    className="button"
    type="button"
  >
    6
  </button>,
  <button
    className="button operator"
    type="button"
  >
    -
  </button>,
  <button
    className="button"
    type="button"
  >
    1
  </button>,
  <button
    className="button"
    type="button"
  >
    2
  </button>,
  <button
    className="button"
    type="button"
  >
    3
  </button>,
  <button
    className="button operator"
    type="button"
  >
    +
  </button>,
  <button
    className="button zero"
    type="button"
  >
    0
  </button>,
  <button
    className="button"
    type="button"
  >
    .
  </button>,
  <button
    className="button operator"
    type="button"
  >
    =
  </button>,
]
`;
